### PR TITLE
Fix back-button browsing and state after mutation events

### PR DESCRIPTION
**Pull request by radiovisual**
Closes issues #20, #21. Adds new dependencies.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-node_modules
+node_modules/*
+!node_modules/github-injection
```
Comment (sindresorhus) [inline .gitignore]: Only include the `index.js` file. Exclude everything else.

File: package.json [hunk 2]
```diff
@@ -9,6 +9,12 @@
     "esnext": true,
     "envs": [
       "browser"
+    ],
+    "ignores": [
+      "deps/*"
```
Comment (sindresorhus) [inline package.json]: This is moot.

**Comment by sindresorhus** [comment]
I can't get it to work:

- Go to https://github.com/sindresorhus/hide-files-on-github
- Click `Show dotfiles`
- Click the `index.js` file.
- Click the browser back button.
- Click the `Hide dotfiles` button.
- *Nothing happens*

**Comment by radiovisual** [comment]
Whoops! I see what is happening. It seems that calling `trigger()` from within the `injector` is not working. The only thing that is working in my latest tests is to re-attach the click handlers to the toggle button upon each injection from within the injector's callback. For whatever reason the injector doesn't want to access the parent scope. I will run some more tests tomorrow and get this fixed.

**Comment by radiovisual** [comment]
Ok, you can try it now. The only way I can get it working after back-button browsing is to continuously attach the click handler. So you will see multiple calls to `addToggleBtnEvents()`. If a single call to this function is removed, the button will lose it's `click` event.

**Comment by sindresorhus** [comment]
Now the button doesn't work on the initial load.

**Comment by radiovisual** [comment]
Hm, that is weird. What is necessary to fix the initial load problem you are mentioning (which just got fixed in the latest push) is not necessary in the options branch I have going for the Options Page. There are a few mysterious forces at work here!

**Comment by sindresorhus** [comment]
I still can't get the button to work on load, but it works in the other PR, so merging. Thank you :)

**Pull request closed without merging**